data/coalition/coalition missions.txt

```diff
@@ -1551,49 +1553,62 @@ mission "Saryd University Lecture"
 			`	"To answer their questions, if you really don't want, I can't stop you, I suppose." Aelbris says with a sigh.`
 					decline
 			label questions
-			`	As soon as you walk up to the stage, several scientists in the crowd stand up out of their chairs. An Arach comes up to you on the stage, and explains that this is their way of asking questions. She selects a Saryd from the crowd.`
+			`	As soon as you walk up to the stage, several scientists in the crowd stand up out of their chairs. The Arach Heliarch comes up to you on the stage, and explains that this is their way of asking questions. She selects a Saryd from the crowd.`
 			branch hai
 				has "First Contact: Hai: offered"
-			`	"Aliens besides the Quarg, are humans in contact with?"`
+			`	"What is the extent of contact that humans have with other aliens?"`
 			choice
-				`	"No, the only aliens humans know about are the Quarg."`
-					goto sad
+				`	"Humans have official contact with the Quarg, but the Quarg don't trade technology with humanity."`
+				`	"Humans have official contact with the Quarg, but humans and Quarg don't really intermix."`
+				`	"Humans know about the Quarg, but most people know little about them."`
+					goto quarg
 			label hai
-			`	"Aliens besides the Quarg, are humans in contact with?"`
+			`	"What is the extent of contact that humans have with other aliens?"`
 			choice
-				`	"Yes, humans are in contact with the Hai. They live to the north of human space, and we can visit them through a wormhole."`
+				`	"Humans are in contact with the Hai. Some of them even live in Hai space. They live to the north of human space, and we can visit them through a wormhole."`
 					goto "know"
 				`	"A few humans know about the Hai, but there's no official acknowledgment of their existence. They live to the north of human space, and we can visit them through a wormhole."`
 					goto "know"
-				`	"No, the only aliens humans know about are the Quarg."`
-					goto "sad"
-			label sad
-			`	At this several in the audience sound disappointed.`
+				`	"Humans have official contact with the Quarg, but the Quarg don't trade technology with humanity."`
+				`	"Humans have official contact with the Quarg, but humans and Quarg don't really intermix."`
+				`	"Humans know about the Quarg, but most people know little about them."`
+					goto "quarg"
+			label quarg
+			`	The audience response is muted. You get the impression that you haven't told them anything they don't already know.`
 				goto nextquestion
 			label know
-			`	Several members of the audience are delighted. "Cooperation between different species, it is important to foster," says the Saryd who asked the question. An Arach asks you to point out where the Hai are on a galactic map, and you do so.`
+			`	Several members of the audience are delighted. "Ah, the Hai, we have heard of," says the Heliarch. "Tell us more of them, could you?"`
```

Comment:
Since the Hai are news to the Coalition, they'd only have heard of them if the player revealed the Hai to them in the First Contact mission. There's no condition that checks for that, though, so an `apply` `set` for something like `"coalition knows hai"` would need to be added to the option where you tell the Heliarchs about the Hai in First Contact, and that would need to be checked here with a branch.
If the player hasn't told them yet, then the condition would be applied here.

Reply:
I tried adding a branch depending on whether or not the Coalition know about the Hai. However, I'm very new to using `apply` `set`, so I might have messed something up.

Reply:
Looks good, however I don't know if `action` works? I thought it was only `apply` that did that, always saw that one be used.

Reply:
I just used what I saw in the wiki: https://github.com/endless-sky/endless-sky/wiki/WritingConversations#action
I think it became a thing in 0.9.15.

Reply:
`Prior to v. 0.9.15 the action node was named apply`
Ah, I see.

In that case just make it so that you also get that condition if you reveal the Hai during the seminar, and should be good.

Reply:
I think I did it now.